### PR TITLE
Update documentation & unit tests for Remind feature

### LINKED ISSUE #116: [PE-D] Reminder pop-up randomly pops up every few minutes even when not touching the app
The reminder window will randomly pop-up even when no input is being made on the application. I am not sure what triggers it, but I did type in the `rm` command once before it starts to happen. Happens every 1-2 minutes.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: WJunHong/ped#1

### LINKED ISSUE #128: [PE-D] Recurring pop-up of reminders
Unable to clear any reminders

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: jaegarpoon/ped#3

### LINKED ISSUE #146: [PE-D] Reminder cleared after reset
A reminder was added back in issue #10. However, now after restarting, it has been cleared

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: WJunHong/ped#11

### LINKED ISSUE #175: Bugs relating to Remind feature
Following are the bugs relating to the Remind feature: 

##

Reminder window contains user with no reminders ❗ 
  - [x] #172 
  - [x] #144 

Reminder tag not displayed in `MainWindow` after adding a reminder ❗  
  - [x] #171 

Placement of Reminder tag in `PersonCard` ❔ 
  - [x] #161 

Reminder for deleted client still retained in Reminders ❗ 
  - [x] #159 
  - [x] #119 

Reminder window doesn't open when Favourite Window is already open ❔ 
  - [x] #155 ([execute `fw` command](https://github.com/jaegarpoon/ped/issues/12), then [execute `rm` command](https://github.com/jaegarpoon/ped/issues/13))

Incorrect title for Favourite Window ❗ 
  - [x] #153 
  - [x] #150 
  - [x] #117 
  - [x] #114 

Reminders not retained after restarting app ❔ -- clarify usage of Reminders in UG 
  - [x] #146 

Unable to clear reminders ❔ -- usage was mentioned in the UG (tester might have missed it) 
  - [x] #128 

Reminder window persistently appears after executing `rm` command ❔ -- clarify behavior of Reminders in UG 
- [x] #116

### PR DESCRIPTION
**Pull request by waffledood**
Changes in this PR: 
- [X] Update unit tests for `RemindCommand` 
- [X] Add unit tests for `ReminderPersons`, `ReminderWindowCommand`, `EditCommand` 
- [X] Update documentation & usage of Reminders in `UserGuide.md` 
- [X] Update `Ui.png` in `index.md` as the previous picture had a reference to the wrong save file `addressbook.json` instead of `realestatepro.json` 

## 
This PR fixes #116 , fixes #128, fixes #146 , fixes #175

### CODE REVIEW

**Review by Ashuh — APPROVED** [review]
LGTM

**Review by cindyangXQ — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 1]
```diff
@@ -328,18 +328,46 @@ Format `viewimage INDEX`
...
+<img src="images/user-guide/successfulRemindCommand.png" height="400px">
+
+Sets a reminder for the specific client from the application. The user (real estate agent) will be able to view a more compact list of clients with reminders set, via a new window called the [Reminder window](#open-reminder-window).
+
+<div markdown="block" class="alert alert-info">
+
+**:information_source: Notes about Reminders:**<br>
+
+- The user will only be able to create 1 Reminder for each client. The user will be able to create a new Reminder, edit a current Reminder or remove a Reminder for a client.
+- Reminders are meant to serve as temporary notes for the user & thus will not persist beyond the lifecycle of a RealEstatePro application. This means any reminders created will be removed once the RealEstatePro application is closed.
```
Comment (hsiaojietng) [inline docs/UserGuide.md:340]: Good that you mention this!

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 2]
```diff
@@ -148,6 +165,113 @@ public void execute_invalidPersonIndexFilteredList_failure() {
         assertCommandFailure(editCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
     }
 
+    @Test
+    public void editBuyerToSeller_success() {
```
Comment (hsiaojietng) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:169]: Naming convention, change to execute_editBuyerToSeller_success().

File: src/test/java/seedu/address/storage/ReminderPersonsTest.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        reminderPersons.setTestingInstance();
+
+        // add Reminder for model Person
+        Person alice = new PersonBuilder().withName("Alice").build();
+        Reminder reminderForAlice = new Reminder("arrange home viewing");
+        reminderPersons.add(alice, reminderForAlice);
+        ReminderPersons.toggleFavouriteForReminder(alice);
+
+        // favoriting a Person
+        Person aliceFavorited = alice.toggleFavourite();
```
Comment (hsiaojietng) [inline src/test/java/seedu/address/storage/ReminderPersonsTest.java:167]: Just a small nit: Favourite rather than favorite

**Review by hsiaojietng — APPROVED** [review]
Just some small nits to change

File: src/test/java/seedu/address/storage/ReminderPersonsTest.java (hunk 3 above)
Comment (waffledood) [reply src/test/java/seedu/address/storage/ReminderPersonsTest.java:167]: thanks for pointing that out! will fix it in another PR

**Pull request merged**